### PR TITLE
a friendly little helper

### PR DESCRIPTION
**Pull request by Kastier1**
### All Submissions:

- [ ] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [ ] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: reflex/model.py [hunk 1]
```diff
@@ -166,10 +166,10 @@ def __init_subclass__(cls):
         non_default_primary_key_fields = [
             field_name
             for field_name, field in cls.__fields__.items()
-            if field_name != "id"
-            and getattr(field.field_info, "primary_key", None) is True
+            if field_name != "id" and sa_column_id_supporter(field)
         ]
         if non_default_primary_key_fields:
+            print(cls.__name__, non_default_primary_key_fields)
```
Comment (masenf) [inline reflex/model.py:172]: Remove print 
```suggestion
```

File: reflex/utils/compat.py [hunk 2]
```diff
@@ -69,3 +69,18 @@ def pydantic_v1_patch():
...
     import sqlmodel as sqlmodel
+
+
+def sa_column_id_supporter(field) -> bool:
+    """Adds support for sa_column so we don't add id with a primary key
+    is defined
+
+    AKA simons_litttle_helper
+    AKA masens_litttlerrr_helperr
+    """
```
Comment (masenf) [inline reflex/utils/compat.py:80]: 😀 docstring must follow repo conventions

File: reflex/utils/compat.py [hunk 3]
```diff
@@ -69,3 +69,18 @@ def pydantic_v1_patch():
 
 with pydantic_v1_patch():
     import sqlmodel as sqlmodel
+
+
+def sa_column_id_supporter(field) -> bool:
```
Comment (masenf) [inline reflex/utils/compat.py:74]: may i suggest

```suggestion
def sqlmodel_field_has_primary_key(field) -> bool:
```

File: reflex/utils/compat.py (hunk 2 above)
Comment (Kastier1) [reply reflex/utils/compat.py]: I believe i have updated to match the repo convention

Comment (masenf) [reply reflex/utils/compat.py]: you need to run pre-commit

```
poetry run pre-commit run --all-files
```

**Review by picklelo — APPROVED** [review]

**Pull request merged**